shared/modules/persistence_module.go

```diff
@@ -33,6 +38,22 @@ type PersistenceModule interface {
 	HandleDebugMessage(*messaging.DebugMessage) error
 }
 
+// TreeStore is fulfilled by the treeStore to create an
+// atomic tree component for use by the peristence context.
```

Comment:
Can you update the comment on the `TreeStore`. I read it and it did not give me much context.

I'm thinking more along the lines of:

`TreeStore provides an interface that guarantees atomic operations (updates and rollbacks) made to the tree supporting the persistence context`